src/v/datalake/arrow_translator.h

```diff
@@ -23,11 +24,10 @@ class struct_converter;
 
 class arrow_translator {
 public:
-    explicit arrow_translator(iceberg::struct_type&& schema);
+    explicit arrow_translator(const iceberg::struct_type& schema);
```

Comment:
i think we can make the consctuctor private and accept plain value, here we accept const ref while in `create` method the schema is moved into `make_optional`